src/app/edit/edit.component.ts

```diff
@@ -17,19 +18,33 @@ export class EditComponent implements OnInit {
   date:any;
   message='';
   editData:any
+  index:any;
+  address:any = [];
+  data:any
   
   constructor(private router:Router,private service: ServicesService) { }
 
   ngOnInit(): void {
+    this.service.getData().subscribe(result=>{
```

Comment:
Please make functions so that code become as much as reusable and invoke them whereever needed